accepted/2.17/enhanced-enums/feature-specification.md

```diff
@@ -64,9 +64,11 @@ It is a **compile-time error** to refer to a declared or default generative cons
 * As the target of a redirecting generative constructor of the same `enum` declaration (`: this(...);`/`: this.targetName(...);`), or
 * Implicitly in the enum value declarations of the same `enum` (`enumValueName(args)`/`enumValueName.targetName(args)`).
 
-_No-one is allowed to invoke a generative constructor and create an instance of the `enum` other than the enumerated enum values. 
-That also means that a redirecting *factory* constructor cannot redirect to a generative constructor of an `enum`,
-and therefore no factory constructor of an `enum` declaration can be `const`, because a `const` factory constructor must redirect to a generative constructor._
+_No-one is allowed to invoke a generative constructor and create an instance of the `enum` other than the enumerated enum values._
+
+It is a **compile-time error** if a declaration of an enumerated value of an `enum` declaration gives rise to an invocation of a factory constructor.
```

Comment:
"Gives rise to" is vague.

Maybe:
> It is a **compile-time error** if the constructor implicitly invoked by an enum value declaration is a factory constructor.

I think this change can be moved into the extension type specification, because it wasn't an issue before.

It should not be breaking, since it was impossible to have an enum value target a non-`const` constructor, or to have a `const` factory constructor which was not redirecting, and a redirecting constructor had to eventually reach a generative constructor of the class, or a subclass. There are no subclasses, and the factory constructor is not allowed to refer to the generative constructors.

(And now there are subtypes with generative constructors which won't necessarily need to create new objects. Another perfect case of breaking previous implicit invariants.)

(But that also means it's not breaking to put it into this spec, since it changes nothing. We just need to be sure implementors see the change as part of implementing extension types.)

Reply:
> .. is vague.

Indeed, now using the proposed text except that I'm using 'enumerated enum value declaration' rather than just 'enum value declaration', in order to remain consistent with other phrases in the vicinity.

> (But that also means it's not breaking to put it into this spec, since it changes nothing. We just need to be sure implementors see the change as part of implementing extension types.)

I think we can ensure that.